### PR TITLE
feat: New tool - Semgrep TAROT-2355 

### PR DESCRIPTION
**Pull request by nicklem**
# ✅ Ready & awaiting feature deployment

Adds Semgrep as a supported tool to the public-facing documentation

### :eyes: Live preview
* [Supported languages and tools](https://tarot-2355-doc-new-tool-documentati--docs-codacy.netlify.app/getting-started/supported-languages-and-tools/)
* [Quality Security Monitor](https://tarot-2355-doc-new-tool-documentati--docs-codacy.netlify.app/repositories/security-monitor/)
* [Configuring code patterns](https://tarot-2355-doc-new-tool-documentati--docs-codacy.netlify.app/repositories-configure/configuring-code-patterns/)

### :construction: To do
- [x] Complete edits to all affected pages
- [x] Complete all subtasks on the relevant Jira ticket ([TAROT-2355](https://codacy.atlassian.net/browse/TAROT-2355))
- [x] Review Semgrep link target ([here](https://github.com/codacy/docs/pull/1921/files#diff-a23aa2ea454fb5eb6067478314f71cd05e09dd1cb53e8276d15171c60a08c606R534), [here](https://github.com/codacy/docs/pull/1921/files#diff-c0ebe10daa0ec96c9164a95452c7f25e4c78029efb9a94db42d9f0c99bf6cf56R245))

[TAROT-2355]: https://codacy.atlassian.net/browse/TAROT-2355?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/repositories/security-monitor.md [hunk 1]
```diff
@@ -74,6 +74,11 @@ The Security Monitor supports checking the languages and frameworks below for an
     docs/repositories-configure/codacy-configuration-file.md (list of tool short names to use on the Codacy configuration file)
 -->
 
+<!-- TODO TAROT-2355
+- Review languages that support security issues. See the task for a complete list.
+- Optional: Flag Semgrep as "beta", e.g. with a footnote mentioning that we only use open-source rules.
```
Comment (nicklem) [inline docs/repositories/security-monitor.md:79]:
```suggestion
- Review languages that support security issues. See the task for a complete list.
```

File: docs/getting-started/supported-languages-and-tools.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 description: List of tools that Codacy uses to analyze over 40 supported languages and frameworks. Codacy provides static analysis for all programming languages and cloud infrastructure-as-code frameworks as well as code duplication, code complexity, and code coverage metrics for most programming languages.
 ---
 
+<!-- TODO TAROT-2355
+- Review extra added languages. See the task for a complete list.
+- Optional: Flag Semgrep as "beta", e.g. with a footnote mentioning that we only use open-source rules.
```
Comment (nicklem) [inline docs/getting-started/supported-languages-and-tools.md:7]:
```suggestion
- Review extra added languages. See the task for a complete list.
```

File: docs/getting-started/supported-languages-and-tools.md [hunk 3]
```diff
@@ -82,11 +83,21 @@ The table below lists all languages and frameworks that Codacy supports and the
       <td>-</td>
       <td>-</td>
     </tr>
+    <tr>
+      <td>Bash</td>
```
Comment (mrfyda) [inline docs/getting-started/supported-languages-and-tools.md:87]: @nicklem  despite Semgrep calling the language Bash, we're mapping it to Codacy's "Shell"

File: docs/repositories-configure/configuring-code-patterns.md [hunk 4]
```diff
@@ -262,6 +262,12 @@ The table below lists the configuration file names that Codacy detects and suppo
     <td><code>scalastyle-config.xml</code>, <code>scalastyle_config.xml</code></td>
     <td></td>
   </tr>
+  <tr>
+    <td>Semgrep</td>
+    <td>Apex, Bash, C++, C#, Dockerfile, Elixir, Go, Java, JavaScript, Kotlin, PHP, Python, Ruby, Rust, Scala, Swift, Terraform, TypeScript</td>
```
Comment (mrfyda) [inline docs/repositories-configure/configuring-code-patterns.md:267]: Bash -> Shell

Comment (mrfyda) [inline docs/repositories-configure/configuring-code-patterns.md:267]:
```suggestion
    <td>Apex, C++, C#, Dockerfile, Elixir, Go, Java, JavaScript, Kotlin, PHP, Python, Ruby, Rust, Scala, Shell, Swift, Terraform, TypeScript</td>
```

File: docs/repositories/security-monitor.md [hunk 5]
```diff
@@ -84,30 +84,38 @@ The Security Monitor supports checking the languages and frameworks below for an
...
+          <a href="https://semgrep.dev/">Semgrep</a> <a href="#semgrep"><sup>1</sup></a></td>
     </tr>
     <tr>
       <td>AWS CloudFormation</td>
       <td><a href="https://github.com/bridgecrewio/checkov/">Checkov</a>,
-          <a href="https://trivy.dev">Trivy</a> <a href="#yaml-only"><sup>1</sup></a></td>
+          <a href="https://trivy.dev">Trivy</a> <a href="#yaml-only"><sup>2</sup></a></td>
+    </tr>
+    <tr>
+      <td>Bash</td>
```
Comment (mrfyda) [inline docs/repositories/security-monitor.md:96]: same here, we can move to Shell

File: docs/getting-started/supported-languages-and-tools.md [hunk 6]
```diff
@@ -505,6 +531,7 @@ The table below lists all languages and frameworks that Codacy supports and the
   </tbody>
 </table>
 
+<sup><span id="semgrep">1</span></sup><!-- TODO TAROT-2355: Review copy and link. -->: Semgrep supports additional security rules when signing up for [Semgrep Pro](https://semgrep.dev/pricing/).
```
Comment (claudiacarpinteiro) [inline docs/getting-started/supported-languages-and-tools.md:534]:
```suggestion
<sup><span id="semgrep">1</span></sup><!-- TODO TAROT-2355: Review copy and link. -->: Semgrep supports additional security rules when signing up for [Semgrep Pro](https://semgrep.dev/pricing/). This tool doesn't support [custom file extensions](../repositories-configure/file-extensions.md).
```

File: docs/repositories-configure/file-extensions.md [hunk 7]
```diff
@@ -6,6 +6,9 @@ description: If your repository has source files with unrecognized extensions, y
 
 If your repository has source files with unrecognized extensions, you can configure Codacy to include them in the next analysis:
 
+!!! note
```
Comment (claudiacarpinteiro) [inline docs/repositories-configure/file-extensions.md:9]: I believe we are excessively highlighting this detail. Consider moving this note to the bottom of the page. And we could add a link to **Semgrep** to the provider's page (https://semgrep.dev/).

Note: Why do we need the `skip-vale` class here? Shouldn't we add the tool name to the [accept.txt](https://github.com/codacy/docs/blob/master/.github/styles/Vocab/Codacy/accept.txt) file instead? 🤔

**Review by claudiacarpinteiro — COMMENTED** [review]
@nicklem I left a couple of suggestions for your evaluation.

**Review by Afonsobli — APPROVED** [review]

**Review by claudiacarpinteiro — APPROVED** [review]
LGTM! 🚀

**Review by mrfyda — APPROVED** [review]

File: docs/repositories/security-monitor.md [hunk 8]
```diff
@@ -215,11 +239,12 @@ The Security Monitor supports checking the languages and frameworks below for an
...
-<sup><span id="spotbugs-plugin">3</span></sup>: Includes the plugin [Find Security Bugs](https://find-sec-bugs.github.io/).  
-<sup><span id="eslint-plugin">4</sup>: Includes the shareable config [<span class="skip-vale">nodesecurity</span>](https://www.npmjs.com/package/eslint-config-nodesecurity) and the plugins [angularjs-security-rules](https://www.npmjs.com/package/eslint-plugin-angularjs-security-rules), [no-unsafe-innerhtml](https://www.npmjs.com/package/eslint-plugin-no-unsafe-innerhtml), [no-unsanitized](https://www.npmjs.com/package/eslint-plugin-no-unsanitized), [scanjs-rules](https://www.npmjs.com/package/eslint-plugin-scanjs-rules), [security](https://www.npmjs.com/package/eslint-plugin-security), and [security-node](https://www.npmjs.com/package/eslint-plugin-security-node).  
-<sup><span id="ruby-31">5</span></sup>: Currently, Codacy doesn't support any static code analysis tool for [Ruby 3.1](https://www.ruby-lang.org/en/news/2021/12/25/ruby-3-1-0-released/).  
+<sup><span id="semgrep">1</span></sup><!-- TODO TAROT-2355: Review copy and link. -->: Semgrep supports additional security rules when signing up for [Semgrep Pro](https://semgrep.dev/pricing/).
```
Comment (nicklem) [inline docs/repositories/security-monitor.md:242]:
```suggestion
<sup><span id="semgrep">1</span></sup>: Semgrep supports additional security rules when signing up for [Semgrep Pro](https://semgrep.dev/pricing/).  
```

**Pull request merged**